src/schema/v2/partner/partner.ts

```diff
@@ -346,8 +349,21 @@ export const PartnerType = new GraphQLObjectType<any, ResolverContext>({
             type: GraphQLString,
           },
         }),
-        resolve: async ({ _id }, args, { partnerSearchCriteriaLoader }) => {
-          if (!partnerSearchCriteriaLoader) return null
+        resolve: async (
+          { _id },
+          args,
+          { partnerSearchCriteriaSingleLoader, partnerSearchCriteriaLoader }
+        ) => {
+          if (
+            !partnerSearchCriteriaLoader ||
+            !partnerSearchCriteriaSingleLoader
+          )
+            return null
+
+          if (args.id && !args.first) {
+            args.first = 1
```

Comment:
Sets `first: 1` behind the scenes if an id argument is present. This makes sure the pagination resolver doesnt 💥 


Let me know if this is ethical? I guess we could return a response without using the `paginationResolver` if were expecting a singular alert pack here from the connection

Reply:
Yah, this was kinda implied by my comment -- wiring up an entirely new field for handling singular items. That said, there are spots all over MP where we do something like this, so seems ok.

Reply:
I actually don't think we want an entirely new field for handling singular items.

Right now - we have this connection type in our schema, where the node is an `Alert`, and the edge contains useful data.

How would a singular field fit into the schema? You might need to create a _new_ type that combines the `Alert` and some other edge data...essentially you'd need to encode the _join_ between partner/alert/criteria 'stuff' - which is a bit hodge-podge and tied to the current backend implementation, yada yada.

For these kinds of situations, I think it's better to not create any singular type, and use the connection of edges (even when you want a single specific edge).